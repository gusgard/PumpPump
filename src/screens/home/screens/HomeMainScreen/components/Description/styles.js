import { StyleSheet } from 'react-native';

import { colors, fontSize } from '@themes';

export default StyleSheet.create({
  description: {
    fontSize: fontSize.medium,
  },
  highlighted: {
    fontWeight: 'bold',
    color: colors.primaryDark,
  },
  readMore: {
    fontWeight: 'bold',
  },
});
