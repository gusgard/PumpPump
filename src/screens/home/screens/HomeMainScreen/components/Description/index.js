import React, { PureComponent } from 'react';
import { Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

const MAX_LINES = 4;

export default class Description extends PureComponent {
  static propTypes = {
    texts: PropTypes.array.isRequired,
  };

  state = {
    readMore: true,
  };

  toggleReadMore = () => {
    this.setState({ readMore: !this.state.readMore });
  };

  render() {
    const { texts } = this.props;
    const { readMore } = this.state;
    return (
      <TouchableOpacity onPress={this.toggleReadMore}>
        <Text numberOfLines={readMore ? MAX_LINES : 0} style={styles.texts}>
          {texts.map(
            (text, index) =>
              text.highlighted ? (
                <Text key={index} style={styles.highlighted}>
                  {text.value}
                </Text>
              ) : (
                <Text key={index}>{text.value}</Text>
              ),
          )}
        </Text>
      </TouchableOpacity>
    );
  }
}
