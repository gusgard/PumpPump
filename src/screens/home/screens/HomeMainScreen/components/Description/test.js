import { shallow } from 'enzyme';
import React from 'react';

import Description from './';

const bio = [
  {
    value: 'Motivation to become the best version of you!  ðŸ’™ðŸ’ªðŸŒŽ. It\'s',
    highlighted: false,
  },
  {
    value: ' #HealthyHolidays',
    highlighted: true,
  },
];

it('renders correctly', () => {
  const wrapper = shallow(<Description bio={bio} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders Description without Read more text', () => {
  const wrapper = shallow(<Description bio={bio} />, {});
  wrapper.simulate('press');
  expect(wrapper).toMatchSnapshot();
});
