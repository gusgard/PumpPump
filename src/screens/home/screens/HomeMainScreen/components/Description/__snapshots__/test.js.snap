// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Description without Read more text 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={0}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      key="0"
    >
      Motivation to become the best version of you!
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      key="1"
      style={
        Object {
          "color": "rgb(15, 45, 90)",
          "fontWeight": "bold",
        }
      }
    >
      #Whitebear
    </Text>
  </Text>
</TouchableOpacity>
`;

exports[`renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={4}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      key="0"
    >
      Motivation to become the best version of you!
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      key="1"
      style={
        Object {
          "color": "rgb(15, 45, 90)",
          "fontWeight": "bold",
        }
      }
    >
      #Whitebear
    </Text>
  </Text>
</TouchableOpacity>
`;
